app/main/views/frameworks.py

```diff
@@ -271,7 +271,15 @@ def framework_submission_services(framework_slug, lot_slug):
         previous_framework = {}
 
     if previous_framework_slug:
-        previous_framework = get_framework_or_404(data_api_client, previous_framework_slug)
+        try:
+            previous_framework = data_api_client.get_framework(previous_framework_slug)['frameworks']
+        except HTTPError as e:
+            current_app.logger.error(
```

Comment:
Unlikely as it is, presumably this could include some kind of transport errors or other failures - so a check that this error is a 404 would be robust.